### ISSUE TITLE
Inter-region peering for Transit Gateways

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#11117]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Transit Gateway now supports the ability to establish peering connections between Transit Gateways in different AWS Regions.

### New or Affected Resource(s)

* `aws_ec2_transit_gateway_peering_attachment`
* `aws_ec2_transit_gateway_peering_attachment_accepter`

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

[Initial announcement](https://aws.amazon.com/about-aws/whats-new/2019/12/aws-transit-gateway-supports-inter-region-peering/).
[Additional regions announcement](https://aws.amazon.com/about-aws/whats-new/2020/04/aws-transit-gateway-now-supports-inter-region-peering-in-11-additional-regions/).
[Blog post](https://aws.amazon.com/blogs/aws/aws-transit-gateway-adds-multicast-and-inter-regional-peering/).
[User guide](https://docs.aws.amazon.com/vpc/latest/tgw/tgw-peering.html).

Requires [AWS SDK v1.25.46](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.46):
* https://github.com/terraform-providers/terraform-provider-aws/pull/11116

**Comment by ewbankkit** [comment]
I'll work on this if there are no objections.

**Comment by j-nix** [comment]
Hi @ewbankkit, myself and @omarimcblack have started working on this at the moment and have transit gateway peering attachments working.

We're now going to write the tests & the accepter for it.

We'll reference the fork/commits here shortly

**Comment by ewbankkit** [comment]
@j-nix Great; Looking forward to it.

**Comment by james1miller93** [comment]
Nice work guys!
We're looking to use this in our environment.

Do you know when it's likely to be released?

**Comment by acarrasquillo** [comment]
@james1miller93 you can track https://github.com/terraform-providers/terraform-provider-aws/pull/11162 and https://github.com/terraform-providers/terraform-provider-aws/pull/11185 for status

**Comment by florianmagnin** [comment]
Hi,
Really need this feature :+1: 
good work

**Comment by skalragg** [comment]
Hi 
Really appreciate that these features are being looked at, is there any estimated timeline when this will be released. I checked #11162 + #11185 but I still am not sure when it will be available.
Thanks

**Comment by bflad** [comment]
Support for the new `aws_ec2_transit_gateway_peering_attachment` resource and data source has been merged, which will release with version 2.60.0 of the Terraform AWS Provider, likely later today. Thanks to @Omarimcblack and @j-nix for the implementation there. 👍 Subscribe to #11162 to receive an automated notification when the release is made. 

Since there is more work to be done to support the `aws_ec2_transit_gateway_peering_attachment_accepter` resource (#11185), we cannot provide an exact estimate for its release, but it should be in the next week or two.

**Comment by bflad** [comment]
Support for the `aws_ec2_transit_gateway_peering_attachment_accepter` resource has been merged and will release shortly with version 2.61.0 of the Terraform AWS Provider. 🎉 

Huge shout out to @Omarimcblack and @j-nix for all their hard work!
